bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/handler/HVACHandler.java

```diff
@@ -47,137 +47,160 @@
 @NonNullByDefault
 public class HVACHandler extends BaseThingHandler {
 
+    /**
+     * The CoolMasterNet protocol's query command returns numbers 0-5 for fan
+     * speed, but the protocol's fan command (and matching binding command) use
+     * single-letter abbreviations.
+     */
+    private static final Map<String, @Nullable String> FAN_NUM_TO_STR;
+
+    /**
+     * The CoolMasterNet query command returns numbers 0-5 for operation modes,
+     * but these don't map to any mode you can set on the device, so we use this
+     * lookup table.
+     */
+    private static final Map<String, @Nullable String> MODE_NUM_TO_STR;
+
+    static {
+        FAN_NUM_TO_STR = new HashMap<>();
+        FAN_NUM_TO_STR.put("0", "l"); // low
+        FAN_NUM_TO_STR.put("1", "m"); // medium
+        FAN_NUM_TO_STR.put("2", "h"); // high
+        FAN_NUM_TO_STR.put("3", "a"); // auto
+        FAN_NUM_TO_STR.put("4", "t"); // top
+
+        MODE_NUM_TO_STR = new HashMap<>();
+        MODE_NUM_TO_STR.put("0", "cool");
+        MODE_NUM_TO_STR.put("1", "heat");
+        MODE_NUM_TO_STR.put("2", "auto");
+        MODE_NUM_TO_STR.put("3", "dry");
+        // 4=='haux' but this mode doesn't have an equivalent command to set it
+        MODE_NUM_TO_STR.put("4", "heat");
+        MODE_NUM_TO_STR.put("5", "fan");
+    }
+
     private final Logger logger = LoggerFactory.getLogger(HVACHandler.class);
 
-    public HVACHandler(Thing thing) {
+    public HVACHandler(final Thing thing) {
         super(thing);
     }
 
+    /**
+     * Get the controller handler for this bridge.
+     *
+     * <p>
+     * This method does not raise any exception, but it will update the Thing
+     * status if there is an issue obtaining the handler.
+     *
+     * <p>
+     * The returned handler may or may not be connected.
+     *
+     * @return the controller handler or null if the controller is unavailable
+     */
     private @Nullable ControllerHandler getControllerHandler() {
-        Bridge bridge = getBridge();
-        return bridge != null ? (ControllerHandler) bridge.getHandler() : null;
+        final Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "CoolMasterNet Controller bridge not set");
+            return null;
+        }
+
+        return (ControllerHandler) bridge.getHandler();
```

Comment:
Should probably also check if the ControllerHandler is null here and change thing status accordingly.